lis/surfacemodels/land/ac.7.2/ac72_readrst.F90

```diff
@@ -0,0 +1,766 @@
+            ! read: WPi
+            call LIS_readvar_restart(ftn, n, LIS_rc%lsm_index, AC72_struc(n)%ac72%WPi, &
+                                    varname="WPi", wformat=wformat)
+
+            ! read: Ziprev
+            call LIS_readvar_restart(ftn, n, LIS_rc%lsm_index, AC72_struc(n)%ac72%Ziprev, &
+                                    varname="Ziprev", wformat=wformat)
+
+            
+            !! integers
+            ! read: Crop_DaysToFullCanopySF
+            call LIS_readvar_restart(ftn, n, LIS_rc%lsm_index, tmptilen_int, &
+                                    varname="Crop_DaysToFullCanopySF", wformat=wformat)
+            do t=1, LIS_rc%npatch(n, LIS_rc%lsm_index)
+                AC72_struc(n)%ac72(t)%Crop%DaysToFullCanopySF = tmptilen_int(t)
+            enddo
+
+            ! read: Crop_GDDaysToFullCanopySF
+            call LIS_readvar_restart(ftn, n, LIS_rc%lsm_index, tmptilen_int, &
+                                    varname="Crop_GDDaysToFullCanopySF", wformat=wformat)
+            do t=1, LIS_rc%npatch(n, LIS_rc%lsm_index)
+                AC72_struc(n)%ac72(t)%Crop%GDDaysToFullCanopySF = tmptilen_int(t)
+            enddo
+
+            ! read: DayLastCut
+            call LIS_readvar_restart(ftn, n, LIS_rc%lsm_index, AC72_struc(n)%ac72%DayLastCut, &
+                                    varname="DayLastCut", wformat=wformat)
+
+            ! read: DayNri
+            call LIS_readvar_restart(ftn, n, LIS_rc%lsm_index, AC72_struc(n)%ac72%DayNri, &
+                                    varname="DayNri", wformat=wformat)
+
+            ! read: DaySubmerged
+            call LIS_readvar_restart(ftn, n, LIS_rc%lsm_index, AC72_struc(n)%ac72%DaySubmerged, &
+                                    varname="DaySubmerged", wformat=wformat)
+
+            ! read: PreviousStressLevel
+            call LIS_readvar_restart(ftn, n, LIS_rc%lsm_index, AC72_struc(n)%ac72%PreviousStressLevel, &
+                                    varname="PreviousStressLevel", wformat=wformat)
+
+            ! read: StressSFadjNEW
+            call LIS_readvar_restart(ftn, n, LIS_rc%lsm_index, AC72_struc(n)%ac72%StressSFadjNEW, &
+                                    varname="StressSFadjNEW", wformat=wformat)
+
+            ! read: SumInterval
+            call LIS_readvar_restart(ftn, n, LIS_rc%lsm_index, AC72_struc(n)%ac72%SumInterval, &
+                                    varname="SumInterval", wformat=wformat)
+
+            !! logical
+            ! read: NoMoreCrop
+            call LIS_readvar_restart(ftn, n, LIS_rc%lsm_index, tmptilen_int, &
+                    varname="NoMoreCrop", wformat=wformat)
+            do t=1, LIS_rc%npatch(n, LIS_rc%lsm_index)
+                if (tmptilen_int(t).eq.1) then
+                    AC72_struc(n)%ac72(t)%NoMoreCrop = .true.
+                else
+                    AC72_struc(n)%ac72(t)%NoMoreCrop = .false.
+                endif            
+            enddo
+
+            !! From PlotVarCrop
```

Comment:
Are these 2 Plot-variables really needed to restart? Either way, I now understand the extent of this "restart" effort. It is insane.

Reply:
Those were tricky ones, you would think they are just meant for plotting but they aren't. To give you an example on how the code is designed: when calling the main function `AdvanceOneTimeStep`, the rooting depth is calculated at the beginning of the day:
`            ! rooting depth at DAP (at Crop.Day1, DAP = 1)
            call SetRootingDepth(AdjustedRootingDepth(GetPlotVarCrop_ActVal(), &
                   GetPlotVarCrop_PotVal(), GetTpot(), GetTact(), &
                   GetStressLeaf(), GetStressSenescence(), &
                   (GetDayNri()-GetCrop_Day1()+1), &
                   GetCrop_DaysToGermination(), &
                   GetCrop_DaysToMaxRooting(), GetCrop_DaysToHarvest(), &
                   GetCrop_GDDaysToGermination(), &
                   GetCrop_GDDaysToMaxRooting(), &
                   GetCrop_GDDaysToHarvest(), GetSumGDDPrev(), &
                   (GetSimulation_SumGDD()), GetCrop_RootMin(), &
                   GetCrop_RootMax(), GetZiprev(), GetCrop_RootShape(), &
                   GetCrop_ModeCycle()))`
                   
You see that the two first arguments are GetPlotVarCrop_ActVal(), GetPlotVarCrop_PotVal() (which were computed on the previous day...)